blockchain/fullblocktests/generate.go

```diff
@@ -2133,3 +2183,134 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 
 	return tests, nil
 }
+
+func GenerateHeaders() (generator *testGenerator, tests [][]TestInstance) {
```

Comment:
We've added two new errors in this PR: `ErrKnownInvalidBlock` and `ErrorMissingParent`. Could you add reject tests for those cases as well?

Reply:
To test these error, we need integrated functionality of header validation and block validation, which is not present right now, I have added tests for these errors in next PR. 
But they are necessary since we are checking for these errors while validating headers.

Reply:
Ok you're correct in that you can't test for `ErrKnownInvalidBlock` until we have the block validation integration but `ErrorMissingParent` is able to be tested currently.

```diff
diff --git a/blockchain/fullblocktests/generate.go b/blockchain/fullblocktests/generate.go
index 45a1a7ed..fdb08830 100644
--- a/blockchain/fullblocktests/generate.go
+++ b/blockchain/fullblocktests/generate.go
@@ -2264,6 +2264,11 @@ func GenerateHeaders() (generator *testGenerator, tests [][]TestInstance) {
        //                                      \-> b1(1)
        g.setTip("b0")
        b1 := g.nextBlock("b1", outs[1])
+
+       // Create a block with the parent as a header that will be rejected.
+       // This block's header will trigger the the error "missing parent".
+       g.nextBlock("b1a", outs[1])
+
        // This can't be done inside a munge function passed to nextBlock
        // because the block is solved after the function returns and this test
        // requires an unsolved block.
@@ -2281,7 +2286,12 @@ func GenerateHeaders() (generator *testGenerator, tests [][]TestInstance) {
                }
                g.updateBlockState("b46", origHash, "b46", b1)
        }
+       g.setTip("b1")
        rejected(blockchain.ErrHighHash)
+
+       g.setTip("b1a")
+       rejected(blockchain.ErrMissingParent)
+
        // Create block with an invalid proof-of-work limit.
        //
        //   ... -> b0()
```

Reply:
Yeah this test will work, I will add this and update the commit.